pkg/datasources/secrets_acceptance_test.go

```diff
@@ -0,0 +1,305 @@
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+const (
+	secretWithClientCredentials      = "snowflake_secret_with_client_credentials"
+	secretWithAuthorizationCodeGrant = "snowflake_secret_with_authorization_code_grant"
+	secretWithBasicAuthentication    = "snowflake_secret_with_basic_authentication"
+	secretWithGenericString          = "snowflake_secret_with_generic_string"
+)
+
+func TestAcc_Secrets_MultipleTypes(t *testing.T) {
+	prefix := random.AlphaN(4)
+	idOne := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "A")
+	idTwo := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "B")
+	idThree := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "C")
+	idFour := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "D")
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "test_oauth_client_id", "test_oauth_client_secret").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "first_scope"}, {Scope: "second_scope"}}),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	refreshTokenExpiryDateTime := time.Now().Add(24 * time.Hour).Format(time.DateTime)
+
+	secretModelBasicAuth := model.SecretWithBasicAuthentication("test", idOne.DatabaseName(), idOne.Name(), "test_passwd", idOne.SchemaName(), "test_username")
+	secretModelGenericString := model.SecretWithGenericString("test", idTwo.DatabaseName(), idTwo.Name(), idTwo.SchemaName(), "foo")
+	secretModelClientCredentials := model.SecretWithClientCredentials("test", integrationId.Name(), idThree.DatabaseName(), idThree.SchemaName(), idThree.Name(), []string{"first_scope", "second_scope"})
+	secretModelAuthorizationCodeGrant := model.SecretWithAuthorizationCodeGrant("test", integrationId.Name(), idFour.DatabaseName(), idFour.SchemaName(), idFour.Name(), "test_token", refreshTokenExpiryDateTime)
+
+	multipleSecretModels := accConfig.FromModel(t, secretModelBasicAuth) +
+		accConfig.FromModel(t, secretModelGenericString) +
+		accConfig.FromModel(t, secretModelClientCredentials) +
+		accConfig.FromModel(t, secretModelAuthorizationCodeGrant) +
+		secretsDataAndVars()
+
+	configVariables := config.Variables{
+		"like": config.StringVariable(prefix + "%"),
+	}
+
+	dsName := "data.snowflake_secrets.test"
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
```

Comment:
Check destroy is missing

Reply:
Is there a way to handle checkDestroy for multiple types? Looking at `TestAcc_SecurityIntegrations_MultipleTypes` there is no check destroy either.

Reply:
I don't think it is. You can write a new function that composes checkDestroys into one.